cpp/src/arrow/ipc/metadata-internal.cc

```diff
@@ -266,7 +266,8 @@ Status MessageBuilder::SetRecordBatch(int32_t length, int64_t body_length,
     const std::vector<flatbuf::FieldNode>& nodes,
     const std::vector<flatbuf::Buffer>& buffers) {
   header_type_ = flatbuf::MessageHeader_RecordBatch;
-  header_ = flatbuf::CreateRecordBatch(fbb_, length, fbb_.CreateVectorOfStructs(nodes),
+  header_ = flatbuf::CreateRecordBatch(fbb_, length, flatbuf::Endianness_Little,
```

Comment:
The best would be to just populate the field based on the underlying Endianness. I've updated the PR